common/src/main/java/net/william278/husktowns/hook/Pl3xMapHook.java

```diff
@@ -0,0 +1,147 @@
+        if (Pl3xMap.api().isEnabled()) Pl3xMap.api().getWorldRegistry().forEach(this::registerLayers);
+
+        plugin.log(Level.INFO, "Enabled Pl3xMap markers hook. Populating web map with claims...");
+        for (World world : plugin.getWorlds()) {
+            plugin.getClaimWorld(world).ifPresent(claimWorld -> setClaimMarkers(claimWorld.getClaims(plugin), world));
+        }
+    }
+
+    @Override
+    public void setClaimMarker(@NotNull TownClaim claim, @NotNull World world) {
+        claims.add(claim);
+    }
+
+    @Override
+    public void removeClaimMarker(@NotNull TownClaim claim, @NotNull World world) {
+        claims.remove(claim);
+    }
+
+    @Override
+    public void setClaimMarkers(@NotNull List<TownClaim> claims, @NotNull World world) {
+        claims.forEach(claim -> setClaimMarker(claim, world));
+    }
+
+    @Override
+    public void removeClaimMarkers(@NotNull List<TownClaim> claims, @NotNull World world) {
+        claims.forEach(claim -> removeClaimMarker(claim, world));
+    }
+
+    @Override
+    public void clearAllMarkers() {
+        claims.clear();
+    }
+
+    @NotNull
+    private String getLayerName() {
+        return plugin.getSettings().getWebMapMarkerSetName();
+    }
+
+    @NotNull
+    private String getClaimMarkerKey(@NotNull TownClaim claim, @NotNull net.pl3x.map.core.world.World world) {
+        return plugin.getKey(
+            claim.town().getName().toLowerCase(),
+            Integer.toString(claim.claim().getChunk().getX()),
+            Integer.toString(claim.claim().getChunk().getZ()),
+            world.getName()
+        ).toString();
+    }
+
+    private void registerLayers(@NotNull net.pl3x.map.core.world.World world) {
+        if (plugin.getSettings().doWebMapHook()) {
+            ClaimsLayer layer = new ClaimsLayer(this, world);
+            world.getLayerRegistry().register(layer);
+        }
+    }
+
+    @EventHandler
+    public void onPl3xMapEnabled(@NotNull Pl3xMapEnabledEvent event) {
+        // Register layers for each world
+        Pl3xMap.api().getWorldRegistry().forEach(this::registerLayers);
+    }
+
+    @EventHandler
+    public void onWorldLoaded(@NotNull WorldLoadedEvent event) {
+        registerLayers(event.getWorld());
+    }
+
+    @EventHandler
+    public void onWorldUnloaded(@NotNull WorldUnloadedEvent event) {
+        event.getWorld().getLayerRegistry().unregister(CLAIMS_LAYER);
+    }
+
+    @NotNull
+    public Options getMarkerOptions(@NotNull TownClaim claim) {
+        return Options.builder()
+            .tooltip(new Tooltip(claim.town().getName()).setDirection(Tooltip.Direction.TOP))
+            // TODO: Unsure what color settings are usually used so left it very bare-bones
+            .fillColor(claim.town().getColor().getRGB())
+            .strokeColor(claim.town().getColor().darker().getRGB())
+            .strokeWeight(1)
+            .build();
```

Comment:
```suggestion
        final Color color = claim.town().getColor();
        return Options.builder()
                .tooltip(new Tooltip(claim.town().getName()).setDirection(Tooltip.Direction.TOP))
                .fillColor(Colors.argb(255 / 2, color.getRed(), color.getGreen(), color.getBlue()))
                .strokeColor(color.darker().getRGB())
                .strokeWeight(1)
                .build();
```
This actually accepts an `argb` int, so we can pass transparency here (as demonstrated), which is nice.

Note that due to #304 my recent push to master deprecates use of java.awt in favor of Adventure -- so we can't use `darker()` anymore, sadly -- when you pull from upstream you will want to go through and correct this.

![image](https://github.com/WiIIiam278/HuskTowns/assets/31187453/0304fe34-a9f4-4ea3-83e2-c7cd9eb5b8ff)